util/src/migration/tests.rs

```diff
@@ -0,0 +1,127 @@
+// Copyright 2015, 2016 Ethcore (UK) Ltd.
+// This file is part of Parity.
+
+// Parity is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Parity is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Parity.  If not, see <http://www.gnu.org/licenses/>.
+
+use std::collections::BTreeMap;
+use migration::{Error, Destination, Migration, Manager, Config};
+
+impl Destination for BTreeMap<Vec<u8>, Vec<u8>> {
+	fn commit(&mut self, batch: BTreeMap<Vec<u8>, Vec<u8>>) -> Result<(), Error> {
+		self.extend(batch);
+		Ok(())
+	}
+}
+
+struct Migration0;
+
+impl Migration for Migration0 {
+	fn from_version(&self) -> &'static str {
+		"0"
+	}
+
+	fn to_version(&self) -> &'static str {
+		"1"
+	}
+
+	fn simple_migrate(&self, key: Vec<u8>, value: Vec<u8>) -> Option<(Vec<u8>, Vec<u8>)> {
+		let mut key = key;
+		key.push(0x11);
+		let mut value = value;
+		value.push(0x22);
+		Some((key, value))
+	}
+}
+
+struct Migration1;
+
+impl Migration for Migration1 {
+	fn from_version(&self) -> &'static str {
+		"1"
+	}
+
+	fn to_version(&self) -> &'static str {
+		"3"
```

Comment:
What's the reason of having `from` and `to` version?
Is it possible to go 3->2?
What is the use case of skipping a number? (like going 1->3 instead of 1->2->3 - what happens if there is this kind of amibguity?)

Wouldn't a `to_version` property for migration be sufficient? (you always go from `to_version-1` and only one way)